src/main/java/functions/BreadthClosureFunction.kt

```diff
@@ -7,11 +7,13 @@ fun breadthClosureFunction(
 ): VisitorFunction {
     return { targetTreeNode ->
         var currentNode = targetTreeNode
-        val answer = mutableListOf<List<TargetTreeNode>>()
-        while (currentNode != null) {
-            answer.add(expression.invoke(currentNode).matches)
+        var currentAnswer = expression.invoke(currentNode)
+        val answer = mutableListOf<List<List<TargetTreeNode>>>(listOf(listOf()))
+        while (currentNode != null && currentAnswer.hasMatch) {
+            answer.add(currentAnswer.matches)
+            currentAnswer = expression.invoke(currentNode)
```

Comment:
I see you also made some changes that weren't named in the PR, such as fixing breadth closure function. I know there aren't any Issues regarding these changes but they could have been named regardless.